src/protocol/libp2p/kademlia/bucket.rs

```diff
@@ -101,12 +101,10 @@ impl KBucket {
 
     /// Get iterator over the k-bucket, sorting the k-bucket entries in increasing order
     /// by distance.
-    pub fn closest_iter<'a, K: Clone>(
-        &'a mut self,
-        target: Key<K>,
-    ) -> impl Iterator<Item = &'a KademliaPeer> {
-        self.nodes.sort_by(|a, b| target.distance(&a.key).cmp(&target.distance(&b.key)));
-        self.nodes.iter().filter(|peer| !peer.addresses.is_empty())
+    pub fn closest_iter<K: Clone>(&self, target: &Key<K>) -> impl Iterator<Item = KademliaPeer> {
```

Comment:
Is `K: Clone` still needed?

Reply:
Yes, as this is the requirement of `Key`.